src/main/java/hudson/plugins/jira/JiraIssueFieldUpdater.java

```diff
@@ -0,0 +1,186 @@
+        return this.issueSelector;
+    }
+
+    @DataBoundSetter
+    public void setIssueSelector(AbstractIssueSelector issueSelector) {
+        this.issueSelector = issueSelector;
+    }
+
+    public String fieldId;
+
+    public String getFieldId() {
+        return this.fieldId;
+    }
+
+    @DataBoundSetter
+    public void setFieldId(String fieldId) {
+        this.fieldId = fieldId;
+    }
+
+    public String fieldValue;
+
+    public String getFieldValue() {
+        return this.fieldValue;
+    }
+
+    @DataBoundSetter
+    public void setFieldValue(String fieldValue) {
+        this.fieldValue = fieldValue;
+    }
+
+    @DataBoundConstructor
+    public JiraIssueFieldUpdater(AbstractIssueSelector issueSelector, String fieldId, String fieldValue) {
+        this.issueSelector = issueSelector;
+        this.fieldId = fieldId;
+        this.fieldValue = fieldValue;
+    }
+
+    public String prepareFieldId(String fieldId) {
+        String prepared = fieldId;
+        if (!prepared.startsWith("customfield_"))
+            prepared = "customfield_" + prepared;
+        return prepared;
+    }
+
+    @Override
+    public void perform(Run<?, ?> run, FilePath workspace, Launcher launcher, TaskListener listener)
+            throws InterruptedException, IOException {
+        PrintStream logger = listener.getLogger();
+
+        AbstractIssueSelector selector = issueSelector;
+        if (selector == null) {
+            logger.println("[JIRA][JiraIssueFieldUpdater] No issue selector found!");
+            throw new IOException("[JIRA][JiraIssueFieldUpdater] No issue selector found!");
+        }
+
+        JiraSite site = JiraSite.get(run.getParent());
+        if (site == null) {
+            logger.println(Messages.Updater_NoJiraSite());
+            run.setResult(Result.FAILURE);
+            return;
+        }
+
+        JiraSession session = null;
+        try {
+            session = site.getSession();
+        } catch (IOException e) {
+            listener.getLogger().println(Messages.Updater_FailedToConnect());
+            e.printStackTrace(listener.getLogger());
+        }
+        if (session == null) {
+            logger.println(Messages.Updater_NoRemoteAccess());
+            run.setResult(Result.FAILURE);
+            return;
+        }
+
+        Set<String> issues = selector.findIssueIds(run, site, listener);
+        if (issues.isEmpty()) {
+            logger.println("[JIRA][JiraIssueFieldUpdater] Issue list is empty!");
+            return;
+        }
+
+        List<JiraIssueField> fields = Lists.newArrayList();
+        fields.add(new JiraIssueField(prepareFieldId(fieldId), fieldValue));
+
+        for (String issue : issues) {
+            submitFields(session, issue, fields, logger);
+        }
+    }
+
+    public void submitFields(JiraSession session, String issueId, List<JiraIssueField> fields, PrintStream logger) {
+        try {
+            session.addFields(issueId, fields);
```

Comment:
If jenkins have no privileges we must notify about this jenkins administrator :) And he will solve the problem. Also i dont like way used by Updater with JiraCarryOverAction by next reasons
- it is not managable for end user
- similar behaviour must be implemented in every similar buildstep
- it is violate the single responsobility principe


I think best way is to create buildstep layer-wrapper to save and rerun not completed Jira* buildstep. We can move JiraCarryOverAction similar behaviour to this layer and this behaviour will work for all similar Jira* buildsteps.